docs/visual-testing/integrations/espresso.md

```diff
@@ -0,0 +1,246 @@
+
+- `buildId(String buildId)`: For advanced users, a user-supplied Sauce Labs Visual build ID. Can be used to create builds in advance using the GraphQL API or CLI tool. This can be used to parallelize tests with multiple browsers, shard, or more. By default, this is not set and we create / finish a build during setup / teardown.
+- `customId(String customId)`: For advanced users, a user-supplied custom ID to identify this build. Can be used in CI to identify / check / re-check the status of a single build. Usage suggestions: CI pipeline ID.
+
+### Build attributes
+
+When creating the service in `VisualClient`, extra fields can be set to define the context, thus acting on which baselines new snapshots will be compared to. ([More info on baseline matching](../../visual-testing.md#baseline-matching))
+
+It needs to be defined through the `VisualApi.Builder` object.
+
+Methods available:
+
+- `buildName(String buildName)`: Sets the name of the build
+- `projectName(String project)`: Sets the name of the project
+- `branchName(String branch)`: Sets the name of the branch
+- `defaultBranchName(String defaultBranch)`: Sets the name of the default branch
+
+Example:
+
+```java
+import com.saucelabs.visual.VisualClient;
+
+visual = VisualClient.builder(username, accessKey)
+          .buildName("Sauce Demo Test")
+          .branchName("main")
+          .projectName("Java examples")
+          .build();
+```
+
+### Ignored regions
+
+#### Component-based ignored region
+
+Sauce Visual provides a way to ignore a list of components.
+
+An ignored component can be a specific element from the page.
+
+Those ignored components are specified when requesting a new snapshot.
+
+Example:
+
+```java
+import com.saucelabs.visual.VisualCheckOptions;;
+
+visualClient.sauceVisualCheck("Inventory Page",
+        VisualCheckOptions.builder()
+                .ignore(withId(R.id.changeTextBt))
+                .build());
+```
+
+#### User-specified ignored region
+
+Alternatively, ignored regions can be user-specified areas. A region is defined by four elements.
+
+- `x`, `y`: The location of the top-left corner of the ignored region
+- `width`: The width of the region to ignore
+- `height`: The height of the region to ignore
+
+_Note: all values are pixels_
+
+You can also give a name to a region that will be visible in Sauce Visual UI using `name`
+
+Example:
+
+```java
+
+import com.saucelabs.visual.VisualCheckOptions;
+import com.saucelabs.visual.model.Region;
+
+visualClient.sauceVisualCheck("Inventory Page",
+        VisualCheckOptions.builder()
+                .ignore(Region.builder()
+                        .name("Demo region")
+                        .x(100)
+                        .y(100)
+                        .width(200)
+                        .height(200)
+                        .build())
+                .build());
+```
+
+### Capturing the DOM snapshot
+
+Sauce Visual does not capture DOM snapshot by default. It can be changed in options.
```

Comment:
"Sauce Visual does not capture DOM snapshots by default. This setting can be changed globally via the build options."